### ISSUE TITLE
instructional text for topic menus 

### ISSUE THREAD

**Issue by SBleeks** [wet-boew/wet-boew#6682]
@wet-boew/access-working-group

Without some instructional text for screen reader users, effectively interacting with the topic menu interface will be unclear for screen reader users.

Can we include something like the following  (offscreen or otherwise)?:
Proposed text for accessing topic menu

Arrow  left and right to move horizontally through the topic menus.  
Arrow up and down to view revolving content within any topics with sub-menus.
Topic menus with no submenu can be treated as a link.  
Pressing TAB leaves the topic menu section.

**Comment by LaurentGoderre** [comment]
Do you have any examples of sites doing this? We are following the ARIA specs for menu which should behave like native menus

**Comment by dfarough** [comment]
I have not ever seen this done however, maybe we could place an expandable summary before the menus which reveals this text so that new users will be able to understand how the menus work.  there is an issue related to expandable help text that discusses this. #6613

**Comment by dfarough** [comment]
Maybe we need to add this feature to all of our templates so that new users will understand these menus and how best to use them.  also it is important to make the purpose of this control obvious so that users will know what to expect when activating or expanding the button.  Probably Menu help would suffice.

**Comment by LaurentGoderre** [comment]
It should be included in the template because that would add extra bloat to all pages. If we are to do this, it needs to be injected by the menu plugin so that it can be cached.

**Comment by dfarough** [comment]
If this is the best way to do this that's fine by me.  However, I gather only certain Menu elements are permitted within ARIA menus so I was more concerned that this information be available without actually having to navigate the menus to find it.  I would not like this control to appear as the last item selection within the menu structure.  If this were the case I could imagine that many users would tab out of the menus without even being aware of this information.  If putting in the menu plugin makes it available to everyone with little cost in performance that's even better.

**Comment by LaurentGoderre** [comment]
We can inject it before the menu very easily. Could you come up with the text for these instructions?

**Comment by dfarough** [comment]
Left and right arrows move across top level menus,
up and down arrows move vertically through menu options 
pressing escape moves focus to the menus top level ,
pressing tab moves out of the menus.

Perhaps this is appropriate for desktop configurations only.

**Comment by SBleeks** [comment]
These are good/brief instructions... I think if we’re going to include some instructional text, we might as well inform the user that the vertical menu items are “revolving”.

I’ll test with Voice Over later today, but imagine that these instructions might not apply for hand held devices.

The point is that for screen reader users, without some sort of instruction, the topic menu interface is not obvious, and thus content will likely be missed.

**Comment by mysticjeff** [comment]
Although, similar commands should work on android and ios with a keyboard.


-----Original Message-----

**Comment by dfarough** [comment]
Yes indicating that pressing down arrow will cycle back to the first menu item again would be good.  I just couldn't think of a good way to say it.  

I like the expandable content way of presenting this text because it allows the user to examine all the options .  this would also be displayed in Braille for Braille users.

**Comment by SBleeks** [comment]
Here’s my original proposed text.  It addresses the revolving vertical menu content:
Arrow left and right to move horizontally through the topic menus.
Arrow up and down to view revolving content within any topics with sub-menus.
Topic menus with no submenu can be treated as a link.
Pressing TAB leaves the topic menu section.

**Comment by LaurentGoderre** [comment]
Is the revolving part really that important? Does not knowing it behaves that way create a barrier?

**Comment by LaurentGoderre** [comment]
I am a bit concerned by the "view revolving content" The content is not revolving, the focus within it wraps when it reach the end. It might be semantic but we don't want to give the illusion that the menu is changing.

**Comment by mysticjeff** [comment]
wraps

-----Original Message-----

**Comment by SBleeks** [comment]
Yes it is important.
Since the items are not in a numbered list, there is no way for blind users to tell when the list item has rolled over.  In some testing we did, users rolled over the list 3 to 4 times before they realized that the list was repeating.  If a simple instruction can improve the experience, why not include it?
Barrier?  Maybe not...
Equal ground... certainly the mention of a revolving vertical menu will at least give screen reader users an indication of what they are dealing with. If the menu list was coded as an html ordered list, this would not be an issue.

**Comment by LaurentGoderre** [comment]
Arrow left and right to move through the topics menus. 
Arrow up and down to move and wrap through sub-topics submenus.

**Comment by SBleeks** [comment]
Definition of revolves: Circle or revolve around another object. Recur in cycles.
Definition of Wrap: Enclose or enfold completely with or as if with a covering. Arrange or coil around.
Definition of loops: a single execution of a set of instructions that are to be repeated. A complete circle in the vertical plane.

Whichever word/description is chosen, it will improve the user experience.  We’re on the right track here.

**Comment by SBleeks** [comment]
1.      Arrow left and right to move horizontally through the topic menus. (addresses the top level menu location and navigational instruction)
2. Arrow up and down to view revolving content within any topics with sub-menus. (addresses the issue of the looping vertical menu items and addresses the navigational instruction.)
3. Topic menus with no submenu can be treated as a link.  (This is the case in the testing we’ve done and would not be known without this instruction.)
4. Pressing TAB leaves the topic menu section. (Addresses the question of how to leave the topic menu section.)

**Comment by LaurentGoderre** [comment]
Maybe wrap is not the best word but your definition of revolving proves my point, it gives the illusion that content is changing (in cycles)

Also is it really that important to say that left and right moves horizontally, that seems redundant to me.

**Comment by dfarough** [comment]
For what it is worth, I like @LaurentGoderre 's suggestion of a menu that raps.  To me revolving suggests a moving target like a rotating carouselle.  we also need to include the escape key which takes you back to the menu.  I don't think that the right and left arrow line is redundant.  For this I think we neeed to be as clear as possible.  I feel that this is primarily  for users who have not experienced such menus before.  The beauty of this presentation is that it does not really get in the way of the menus and that it would be possible to remove the verbal instructions which occur when the menu itself is opened.  These types of menus are occurring more often these days so providing this type of instruction each time the menu opens will become less necessary in  time.

**Comment by jrowsell** [comment]
This sounds good.
Thanks for the research Stacey!

**Comment by dfarough** [comment]
Perhaps circular is a better word to describe the topic menus.

**Comment by LaurentGoderre** [comment]
technically, it's also the top level menu that have that behavior so we could say:

1. Arrow left and right to cycle horizontally through the topic menus. 
2. Arrow up and down to cycle within any topics with sub-menus.

**Comment by dfarough** [comment]
what do you think of the following text?

1.      Arrow left and right to move horizontally through the topic menus. 
2. Arrow up and down to view circular  content within any topics with sub-menus. 
3. Topic menus with no submenu can be treated as a link.   
4. Pressing TAB leaves the topic menu section. 
5. Pressing escape returns focus to the focussed top level menu.

**Comment by LaurentGoderre** [comment]
I would rather the verb be the focus action so that it's clear that it's the focus that moves not the content

**Comment by SBleeks** [comment]
Yes! “cycle” is perfect!

**Comment by dfarough** [comment]
Earlier we were discussing the need to make it obvious that the submenus were circular.  this is less obvious using your text.

A blind person does not see the focus move from the bottom of the menu back up to the first menu item.  The only way for a blind user to know for sure that the   final menu item was reached is to hear the first item spoken again.  To me this is not obvious using this text.

Grouping it that way makes sense to me though.

**Comment by LaurentGoderre** [comment]
Shouldn't the aria-pos announce when the menu goes back to the first item? I think that with the cycle as the action plus the ARIA it might be exactly what we need.

**Comment by LaurentGoderre** [comment]
I would also like to know what @mysticjeff and @jrowsell think about it too.

**Comment by SBleeks** [comment]
@LaurentGoderre I like your wording:
Definition of cycle: An interval during which a recurring sequence of events occurs.

 1.  Arrow left and right to cycle through the topic menus.
 2.  Arrow up and down to cycle  through any topics with sub-menus.

@dfarough do you think this describes the required behaviour and offers enough information/instruction to improve the user experience?
We had previously listed these 3 points:

 1.  Topic menus with no submenu can be treated as a link.
 2.  Pressing TAB leaves the topic menu section.
 3.  Pressing escape returns focus to the focussed top level menu.

I believe the “expand/ collapse” help is ideal.  This way the instructional help text is adjacent to the plugin and for those users who have become familiar with the design, they don’t have to see/hear it everytime as would be the case if the text were simply hidden/ offscreen.

—

**Comment by SBleeks** [comment]
We’re getting there.

**Comment by dfarough** [comment]
@SBleeks   I think that the other items mentioned should be included.  especially with regard to the tab key. Users should be told about this so that if they want to get out of the menus quickly that they can do it easily.  the mention of the escape key might be considered redundant but it is always a good idea to provide several ways to perform a task.  I don't think it  is necessary to make this text unnecessarily brief.  We should try to anticipate as many situations as possible without going into great detail.  I like the idea of a list of operations like we are doing now.

**Comment by SBleeks** [comment]
Agreed.

**Comment by mysticjeff** [comment]
This all sounds good to me.

-----Original Message-----

**Comment by jrowsell** [comment]
Yes, and yes aria-pos SHOULD read position, but more inclusive info such as what is below works best.

**Comment by LaurentGoderre** [comment]
Ok, so here is a recap:

1. Arrow left and right to cycle horizontally through the topic menus.
2. Arrow up and down to cycle within any topics with sub-menus.
3. Pressing escape returns focus to the focussed top level menu.
4. Topic menus with no submenu can be treated as a link.
5. Pressing TAB leaves the topic menu section.

**Comment by LaurentGoderre** [comment]
I am still unsure about number 3 and 4

For number 4 I would suggest: "Topic menus items with no submenu can be treated as a link." (I added the word items)

Number 3 sounds awkward return focus to the focussed

**Comment by SBleeks** [comment]
Excellent!  As you said, the word “horizontally” is probably redundant here.

**Comment by SBleeks** [comment]
Yes, remove the word “focus” as it is redundant here.
Adding the word “item” does improve context.
I’ve slightly reordered the points so they flow, and incorporated our wording changes. How’s this?:


 1.  Arrow left and right to cycle through the topic menus.
 2.  Arrow up and down to cycle within any topics with sub-menus.

3.      Topic menu items  with no submenu can be treated as a link.

4.      Pressing escape returns to the top level menu.

5.      Pressing TAB leaves the topic menu section.
—

**Comment by LaurentGoderre** [comment]
Nice!!! Does everyone agree with this?

**Comment by dfarough** [comment]
@SBleeks That's a wrap!

**Comment by dfarough** [comment]
I agree! Just in case there is any confusion.

**Comment by mysticjeff** [comment]
Works for me 

-----Original Message-----

**Comment by SBleeks** [comment]
@dfarough, Nice wrap! LOL Wrap it up! It’s good. Best rap I’ve heard since David MacDonald’s WCAG youtube rap!

**Comment by jrowsell** [comment]
This works ☺

**Comment by LaurentGoderre** [comment]
Ok, now to get it translated. @thomasgohard how do we usually do this?

**Comment by thomasgohard** [comment]
If it's short, I usually do the translation myself.

Is this the text you need translated?
> 1. Arrow left and right to cycle through the topic menus.
> 2. Arrow up and down to cycle within any topics with sub-menus.
> 3. Topic menu items  with no submenu can be treated as a link.
> 4. Pressing escape returns to the top level menu.
> 5. Pressing TAB leaves the topic menu section.

**Comment by LaurentGoderre** [comment]
Yep

**Comment by thomasgohard** [comment]
You should always start list items with the same type of word (e.g., all verbs or all nouns). Also, active voice is better for readability and comprehension. As these are instructions, they should start with a verb. I also found the wording of 2 and 4 confusing.

Any problems with editing as follows?
1. Use the left and right arrows to cycle through the topics menu.
2. Use the up and down arrows to cycle through a topic's sub-menu.
3. Treat topics with no sub-menus as links.
4. Press escape to close an open sub-menu.
5. Press tab to leave to topics menu.

**Comment by dfarough** [comment]
This works for me.  I like the active voice for instructions.

**Comment by thomasgohard** [comment]
Here's the French:
1. Utiliser les flèches vers la gauche et la droite pour faire défiler le menu des sujets.
2. Utiliser les flèches vers le haut et le bas pour faire défiler le sous-menu d'un sujet.
3. Traiter les sujets sans sous-menu comme des liens.
4. Appuyer sur la touche d'échappement pour fermer un sous-menu qui est ouvert.
5. Appuyer sur la touche de tabulation pour quitter le menu des sujets.

**Comment by LaurentGoderre** [comment]
@crichard is this something you can help out with?

**Comment by crichard** [comment]
I can't this week, traveling for work.  I can look at this next week if it's still open.

**Comment by LaurentGoderre** [comment]
Absolutely! Thank you!

**Comment by SBleeks** [comment]
For point #2, the idea of “cycle through the topics” is missed.  Maybe the term is “”en bouqule”?

**Comment by thomasgohard** [comment]
@SBleeks "faire défiler" means "to cycle through," "en boucle" means "in a continuous loop."

**Comment by SBleeks** [comment]
Ah bien... Fantastique!

**Comment by LaurentGoderre** [comment]
@thomasgohard Can you add these to the i18n spreadsheet?

**Comment by hsrudnicki** [comment]
@thomasgohard @LaurentGoderre @SBleeks and others: Are you ready to close this issue?

**Comment by SBleeks** [comment]
yes. I think this issue can be closed with one minor modification.  There is a typo for #5.
"5. Press tab to leave to topics menu." Should be
"5. Press tab to leave the topics menu"

Simply replace "to" with 'the".

Thanks.

**Comment by LaurentGoderre** [comment]
No this can't be closed it was not implemented yet

**Comment by hsrudnicki** [comment]
Made "Going Stale" to close this issue as resolved/fixed due to the above milestone set or if otherwise? Either way, please update this issue's status, thank you

**Comment by hsrudnicki** [comment]
Please update this issue's status/milestone in regards to the current year, thanks

**Comment by hsrudnicki** [comment]
Does this remain still an open issue?
